src/libstore/local-fs-store.cc

```diff
@@ -61,9 +61,9 @@ struct LocalStoreAccessor : public FSAccessor
         return res;
     }
 
-    std::string readFile(const Path & path) override
+    std::string readFile(const Path & path, bool requireValidPath = true) override
```

Comment:
I think the overrides shouldn't have the default args? Elsewhere in the codebase they at least are only part of the declarations, IIRC.

Reply:
That's a good question. Triggered me to read https://stackoverflow.com/questions/6464404/virtual-function-default-arguments-behaviour which is… interesting.

The gist of it seems to that if we only call `accessor.readFile(path)` where `accessor` is declared of type `FSAccessor` (which is probably the case) then we don't need the default arguments for the overrides. But that looks quite fragile. Maybe I could just split the interface as:

```cpp
virtual bool isValidPath(const Path& path);
virtual std::string readFileUnchecked(const Path& path);
std::string readFile(const Path& path) {
  if (!isValidPath(path)) throw InvalidPath(...);
  return readFileUnchecked(path);
}
```

Reply:
Hmm I read it the other way around, where up casting certainly gives you the outer default arg, but no upcasting would also give you the outer default arg if there is no inner default arg.

Reply:
Nah, just checked:

```cpp
struct Foo {
  virtual void foo(int n = 1) { }
};

struct Bar : Foo {
  void foo(int n) override { }
};

int main() {
  Foo x;
  x.foo();
  Bar y;
  y.foo();
}
```

`x.foo()` typechecks fine, but `y.foo()` doesn't

Reply:
Sorry to have misled you with that wrong interpretation then.